homeassistant/components/ssdp/__init__.py

```diff
@@ -426,10 +421,18 @@ async def async_start(self) -> None:
     async def _async_start_ssdp_listeners(self) -> None:
         """Start the SSDP Listeners."""
         for source_ip in await self._async_build_source_set():
+            source_ip_str = str(source_ip)
+            source_tuple: AddressTupleVXType = (
+                (source_ip_str, 0, 0, int(getattr(source_ip, "scope_id")))
+                if source_ip.version == 6
+                else (source_ip_str, 0)
+            )
```

Comment:
Usually we break up ternaries if they span multiple lines

Reply:
What do you propose? When I convert it to `if ...: source_tuple = ... else: source_tuple = ...` mypy complains about `source_tuple` already being defined. I think this ternary is more clean than the if-version or something like `source_tuple = source_tuple_v6 or source_tuple_v4`.